### ISSUE TITLE
[Bug]: When using Apache with mod-php, a SEGV occurs after starting Apache for a while.

### ISSUE THREAD

**Issue by namazu510** [DataDog/dd-trace-php#2266]
### Bug report

I am using mod-php with Apache. For mpm, I use prefork. After starting Apache, the application works fine for a while, but after some time, Apache experiences a SEGV. I tried settings like opcache.enable = 0, but the situation did not improve. Disabling the dd-trace extension resolves the issue.

coredump info
```
Core was generated by `/usr/sbin/apache2 -k start'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  ddtrace_open_span (span=span@entry=0x7fdfec2523c0) at /home/circleci/datadog/tmp/build_extension/ext/span.c:134
134	/home/circleci/datadog/tmp/build_extension/ext/span.c: No such file or directory.
[Current thread is 1 (Thread 0x7fdfef915d40 (LWP 2282168))]
(gdb) backtrace
#0  ddtrace_open_span (span=span@entry=0x7fdfec2523c0) at /home/circleci/datadog/tmp/build_extension/ext/span.c:134
#1  0x00007fdfeb7ea86d in ddtrace_push_root_span () at /home/circleci/datadog/tmp/build_extension/ext/span.c:360
#2  0x00007fdfeb7da082 in dd_initialize_request () at /home/circleci/datadog/tmp/build_extension/ext/ddtrace.c:815
#3  0x00007fdfeb7da1b5 in ddtrace_alter_dd_trace_disabled_config (old_value=<optimized out>, new_value=new_value@entry=0x7ffebfe5b430) at /home/circleci/datadog/tmp/build_extension/ext/ddtrace.c:974
#4  0x00007fdfeb7b748c in ZaiConfigOnUpdateIni (entry=0x55b694ca5640, new_value=0x7fdfec26f040, mh_arg1=<optimized out>, mh_arg2=<optimized out>, mh_arg3=<optimized out>, stage=<optimized out>) at /home/circleci/datadog/tmp/build_extension/zend_abstract_interface/config/config_ini.c:195
#5  0x00007fdfecb476dd in zend_alter_ini_entry_ex () from /usr/lib/apache2/modules/libphp.so
#6  0x00007fdfeb7b832d in zai_config_ini_rinit () at /home/circleci/datadog/tmp/build_extension/zend_abstract_interface/config/config_ini.c:386
#7  0x00007fdfeb7b6794 in zai_config_rinit () at /home/circleci/datadog/tmp/build_extension/zend_abstract_interface/config/config.c:152
#8  0x00007fdfeb7d897f in ddtrace_activate () at /home/circleci/datadog/tmp/build_extension/ext/ddtrace.c:273
#9  0x00007fdfecac8c7e in zend_llist_apply () from /usr/lib/apache2/modules/libphp.so
#10 0x00007fdfecac5086 in ?? () from /usr/lib/apache2/modules/libphp.so
#11 0x00007fdfecad60b3 in zend_activate () from /usr/lib/apache2/modules/libphp.so
#12 0x00007fdfeca6fb21 in php_request_startup () from /usr/lib/apache2/modules/libphp.so
#13 0x00007fdfecbb84c8 in ?? () from /usr/lib/apache2/modules/libphp.so
#14 0x000055b6948449d0 in ap_run_handler ()
#15 0x000055b694844fc6 in ap_invoke_handler ()
#16 0x000055b69485cd7e in ap_internal_redirect ()
#17 0x00007fdfed85f428 in ?? () from /usr/lib/apache2/modules/mod_rewrite.so
#18 0x000055b6948449d0 in ap_run_handler ()
#19 0x000055b694844fc6 in ap_invoke_handler ()
#20 0x000055b69485d9db in ap_process_async_request ()
#21 0x000055b69485dc1e in ap_process_request ()
#22 0x000055b694859a94 in ?? ()
#23 0x000055b69484e730 in ap_run_process_connection ()
#24 0x00007fdfef839bbc in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#25 0x00007fdfef839f26 in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#26 0x00007fdfef839f71 in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#27 0x00007fdfef83a67f in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#28 0x000055b694824820 in ap_run_mpm ()
#29 0x000055b69481c148 in main ()
```

Apache Version:
```
Server version: Apache/2.4.56 (Debian)
```

OS Version:
```
Linux XXX 5.10.0-23-amd64 #1 SMP Debian 5.10.179-1 (2023-05-12) x86_64 GNU/Linux
```

I am always greatly helped by DataDog, thank you.

### PHP version

8.2.5

### Tracer or profiler version

0.91.1

### Installed extensions

[PHP Modules]
apcu
bcmath
bz2
calendar
Core
ctype
curl
date
ddtrace
dom
exif
fileinfo
filter
gd
hash
iconv
imagick
intl
json
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
random
redis
Reflection
session
SimpleXML
sockets
solr
SPL
sqlite3
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xml
xmlreader
xmlwriter
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
ddtrace

### Output of `phpinfo()`

<img width="818" alt="image" src="https://github.com/DataDog/dd-trace-php/assets/11805741/6fa92f0e-9f4f-402a-855b-44fa0f58d9d2">
<img width="892" alt="image" src="https://github.com/DataDog/dd-trace-php/assets/11805741/a38db220-32ee-45fb-8603-032afc9a5d22">
<img width="766" alt="image" src="https://github.com/DataDog/dd-trace-php/assets/11805741/307bee89-d8e6-4913-a715-cd4ce668baa7">
<img width="819" alt="image" src="https://github.com/DataDog/dd-trace-php/assets/11805741/ca662ab9-ffcd-4e0f-8f81-c07bc75f158e">
<img width="788" alt="image" src="https://github.com/DataDog/dd-trace-php/assets/11805741/2d3d358a-4bf2-411d-83ba-5f279b893e72">

### Upgrading from

I am upgrading from PHP 7.3.5 and ddtrace v0.48.3. I've tried several versions of ddtrace, but it seems that the versions that are compatible with PHP 8.2 have this issue.

**Comment by bwoebi** [comment]
Hey @namazu510,

we've identified that particular issue already and are in progress of fixing it. It will be fixed soon :-)

**Comment by bwoebi** [comment]
Hey @namazu510, this has been released just now as 0.92.0 :-)
